core/src/main/scala/chisel3/DataImpl.scala

```diff
@@ -470,21 +470,32 @@ private[chisel3] trait DataImpl extends HasId with NamedComponent { self: Data =
     }
   }
 
+  // Specializes the .toString method of a [[Data]] for conditions such as
+  //  DataView, Probe modifiers, a DontCare, and whether it is bound or a pure chisel type
   private[chisel3] def stringAccessor(chiselType: String): String = {
+    // Add probe and layer color (if they exist) to the returned String
+    def addProbeModifier(chiselType: String): String = {
+      probeInfo match {
+        case None => chiselType
+        case Some(ProbeInfo(writeable, layer)) =>
+          val layerString = layer.map(x => s"[${x.fullName}]").getOrElse("")
+          (if (writeable) "RWProbe" else "Probe") + s"$layerString<$chiselType>"
+      }
+    }
```

Comment:
```suggestion
    val chiselTypeWithModifier =
      probeInfo match {
        case None => chiselType
        case Some(ProbeInfo(writeable, layer)) =>
          val layerString = layer.map(x => s"[${x.fullName}]").getOrElse("")
          (if (writeable) "RWProbe" else "Probe") + s"$layerString<$chiselType>"
      }
```
Then use `chiselTypeWithModifier` below anywhere `chiselType` was being used.